tests/unit_tests/drivers/test_jobscript_driver.py

```diff
@@ -62,72 +160,121 @@ def fixture_job_options(tmp_path, job_id, experiment_name, injected_input_files)
-
-    # add both parameters to the template
-    input_template_2 = tmp_path / "input_template_2.yaml"
-    templatify(parameters.names + ["input_1", "experiment_name"], input_template_2)
-
-    return input_template_1, input_template_2
-
-
-@pytest.fixture(name="injected_input_files")
-def fixture_injected_input_files(tmp_path, job_id, experiment_name):
-    """Fixture for the create input files."""
-    input_file_1 = tmp_path / str(job_id) / f"{experiment_name}_input_1_{job_id}.yaml"
-    input_file_2 = tmp_path / str(job_id) / f"{experiment_name}_input_2_{job_id}.yaml"
-    injected_input_files = {"input_1": input_file_1, "input_2": input_file_2}
-
-    return injected_input_files
-
-
 @pytest.fixture(name="jobscript_driver")
-def fixture_jobscript_driver(parameters, input_templates):
+def fixture_jobscript_driver(parameters, input_templates, executable):
     """Jobscript driver object."""
     input_template_1, input_template_2 = input_templates
 
     driver = JobscriptDriver(
         parameters=parameters,
         jobscript_template="",
-        executable="",
+        executable=executable,
         input_templates={"input_1": input_template_1, "input_2": input_template_2},
     )
 
     return driver
 
 
-def test_jobscript_driver_multiple_input_files(
-    jobscript_driver, job_options, injected_input_files, parameters
+@pytest.fixture(name="args_init")
+def fixture_args_init(
+    parameters,
+    jobscript_template,
+    executable,
+    input_template,
+    files_to_copy,
+    data_processor,
+    gradient_data_processor,
+    jobscript_file_name,
+    extra_options,
 ):
+    """Arguments to initialize a JobscriptDriver.
+
+    These arguments are meant for initialization with the default
+    constructor.
+    """
+    # pylint: disable=unused-argument
+    args_init = locals()
+    args_init["input_templates"] = args_init["input_template"]
+    args_init.pop("input_template")
+    return args_init
+
+
+def assert_jobscript_driver_attributes(jobscript_driver, args_init):
+    """Assert that the jobscript driver attributes are set correctly."""
+    args_init["files_to_copy"].append(args_init["input_templates"])
+
+    assert jobscript_driver.parameters == args_init["parameters"]
+    assert jobscript_driver.input_templates == {"input_file": args_init["input_templates"]}
+    assert jobscript_driver.jobscript_template == args_init["jobscript_template"]
+    assert jobscript_driver.jobscript_options["executable"] == args_init["executable"]
+    assert jobscript_driver.files_to_copy == args_init["files_to_copy"]
+    assert jobscript_driver.data_processor == args_init["data_processor"]
+    assert jobscript_driver.gradient_data_processor == args_init["gradient_data_processor"]
+    assert jobscript_driver.jobscript_file_name == args_init["jobscript_file_name"]
+    assert jobscript_driver.jobscript_options.items() >= args_init["extra_options"].items()
```

Comment:
Why not check `==`? Maybe I am missing something but since we are testing I guess we should check it exactly